packages/lib/shared/components/btns/button-group/ButtonGroup.tsx

```diff
@@ -1,66 +1,141 @@
-                  borderRadius="4px"
-                  inset="0"
-                  layoutId={`active-${groupId}`}
-                  position="absolute"
-                  shadow="md"
-                />
-              )}
-              <Box position="relative" zIndex="8">
-                {option.label}
-              </Box>
-            </Button>
+          return option?.tabTooltipLabel ? (
+            <Box flex="1" key={`button-group-option-${option.value}`}>
+              <Popover trigger="hover">
+                <PopoverTrigger>
+                  <Box _hover={{ opacity: 0.75 }} transition="opacity 0.2s var(--ease-out-cubic)">
+                    <GroupOptionButton isActive={isActive} option={option} {...props} />
+                  </Box>
+                </PopoverTrigger>
+                {option?.tabTooltipLabel && (
+                  <PopoverContent maxW="300px" p="sm" w="auto">
+                    <Text fontSize="sm" variant="secondary">
+                      {option.tabTooltipLabel}
+                    </Text>
+                  </PopoverContent>
+                )}
+              </Popover>
+            </Box>
+          ) : (
+            <Box flex="1" key={`button-group-option-${option.value}`}>
+              <GroupOptionButton
+                isActive={isActive}
+                key={`button-group-option-${option.value}`}
+                option={option}
+                {...props}
+              />
+            </Box>
           )
         })}
       </HStack>
     </LayoutGroup>
   )
 }
+
+function GroupOptionButton({
+  option,
+  isActive,
+  size,
+  width,
+  groupId,
+  hasLargeTextLabel,
+  onChange,
+}: { option: ButtonGroupOption; isActive: boolean } & Props) {
+  return (
+    <Button
+      bg="transparent"
+      id={`button-group-${option.value}`}
+      isDisabled={option.disabled}
+      key={`button-group-option-${option.value}`}
+      onClick={() => onChange(option)}
+      position="relative"
+      rightIcon={<IconPopover option={option} />}
+      role="group"
+      size={size}
+      variant={isActive ? 'buttonGroupActive' : 'buttonGroupInactive'}
+      width={width || 'full'}
+    >
+      {isActive && (
+        <Box
+          as={motion.div}
+          bg="background.button.secondary"
+          borderRadius="4px"
+          inset="0"
+          layoutId={`active-${groupId}`}
+          position="absolute"
+          shadow="md"
+        />
+      )}
+      <Box fontSize={hasLargeTextLabel ? 'lg' : undefined} position="relative" zIndex="8">
+        {option.label}
+      </Box>
+    </Button>
+  )
+}
+
+function IconPopover({ option }: { option: ButtonGroupOption }) {
+  if (option.tabTooltipLabel) return null
+  if (!option.iconTooltipLabel) return null
+  return (
+    <Popover key={option.iconTooltipLabel} placement="top" trigger="hover">
+      <PopoverTrigger>
+        <Icon as={Info} />
+      </PopoverTrigger>
+      {option?.iconTooltipLabel && (
```

Comment:
the condition here is not needed i think

Reply:
Removed